orttraining/orttraining/python/ort_trainer.py

```diff
@@ -648,6 +648,10 @@ def _init_session(self):
                 *self.model_desc_.outputs_,
                 IODescription(get_all_gradients_finite_arg_name(self.session), [1], torch.bool)]
 
+        if self.state_dict_:
```

Comment:
In [Pytorch API](https://pytorch.org/tutorials/beginner/saving_loading_models.html#save-load-state-dict-recommended), users are responsible for loading the dict after the model is recreated. Should we use the same approach?

```
model = TheModelClass(*args, **kwargs)
model.load_state_dict(torch.load(PATH))
```

Reply:
Yea, makes sense. The load_state_dict(...) call failed in the BERT example because ORTTrainer(...) does not create ONNX model or ORT session when passed a torch.nn.Module. It only does this when it receives sample inputs during evaluation or training. Should we be passing sample input on construction?

Reply:
After reviewing one more time, I agree this code works fine!
A pseudo code is:
```python
model = ORTTrainer(...) #  no state dict to load
model.load_state_dict(torch.load(PATH)) # state dict is saved for delayed load
...
for batch in train_loader:
    model.train_step(batch) # training session and a state dict exist. state is loaded
    ...
```